builtin/providers/azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -735,19 +763,25 @@ func flattenAzureRMVirtualMachineScaleSetOsProfile(profile *compute.VirtualMachi
 
 func flattenAzureRmVirtualMachineScaleSetStorageProfileOSDisk(profile *compute.VirtualMachineScaleSetOSDisk) []interface{} {
 	result := make(map[string]interface{})
-	result["name"] = *profile.Name
```

Comment:
This one is a little tricky. Technically, `name` is a optional field in the [schema](https://github.com/Azure/azure-resource-manager-schemas/blob/master/schemas/2016-04-30-preview/Microsoft.Compute.json#L2277) but it is required for vhd_containers and images. If `name` is defined with a `managed_disk_type`, it gives error `Status=400 Code="InvalidParameter" Message="Parameter 'osDisk.name' is not allowed."`
Please let me know which one is preferred, leaving it as is or changing it to a an optional field and using validation tests.